### PR TITLE
Fix parsing of <arg> syntax in scenarios

### LINKED ISSUE #1799: Regular sections recognize <arg> syntax
```c++
SECTION("HTML uses <tag> syntax") {
  // crashes, because CAF tries to parse `<tag>` as-if using `OUTLINE`
}
```

This boils down to the parsing in https://github.com/actor-framework/actor-framework/blob/master/libcaf_test/caf/test/block.cpp#L98. This is currently unconditional. Instead, we need to introduce a new `block_type` for `outline` and only do the parsing for this type.

### PR DESCRIPTION
**Pull request by riemass**
Closes #1799.
Decided to do this first because having an dedicated outline block might be useful for #1776.

### CODE REVIEW

File: libcaf_test/caf/test/block.cpp [hunk 1]
```diff
@@ -102,49 +102,55 @@ void block::lazy_init() {
...
-            state = cpy_state::verbatim;
-            break;
-          default:
-            parameter_name += c;
-            break;
-        }
-        break;
+  // Process <arg> syntax if the root block is an outline.
+  if (!ctx_->call_stack.empty()
+      && ctx_->call_stack.front()->type() == block_type::outline) {
```
Comment (Neverlord) [inline libcaf_test/caf/test/block.cpp:107]: Prefer guard style:

```suggestion
  if (ctx_->call_stack.empty()
      || ctx_->call_stack.front()->type() != block_type::outline) {
    description_ = raw_description_;
    return;
  }
```

File: libcaf_test/caf/test/runnable.cpp [hunk 2]
```diff
@@ -29,6 +30,14 @@ void runnable::run() {
   auto guard
     = detail::scope_guard{[]() noexcept { current_runnable = nullptr; }};
   switch (root_type_) {
+    case block_type::outline:
+      if (auto guard = ctx_->get<outline>(0, description_, loc_)->commit()) {
+        do_run();
+        return;
+      }
+      CAF_RAISE_ERROR(std::logic_error,
+                      "failed to select  the root block for the scenario");
```
Comment (Neverlord) [inline libcaf_test/caf/test/runnable.cpp:39]:
```suggestion
                      "failed to select  the root block for the outline");
```

File: libcaf_test/caf/test/scenario.test.cpp [hunk 3]
```diff
@@ -25,6 +25,23 @@ SCENARIO("a scenario may not contain a section") {
...
+  GIVEN("given a <description>") {
+    WHEN("when <parsing> the raw description") {
+      THEN("then we copy the <description> verbatim") {
+        auto i = ctx_->call_stack.begin();
+        check_eq((*i++)->description(),
+                 "scenario with <foo> syntax in description");
+        check_eq((*i++)->description(), "given a <description>");
+        check_eq((*i++)->description(), "when <parsing> the raw description");
+        check_eq((*i++)->description(),
+                 "then we copy the <description> verbatim");
```
Comment (Neverlord) [inline libcaf_test/caf/test/scenario.test.cpp:40]: Unsafe, and IMHO a bit hard to follow due to the iterator indirections. I'd prefer something like this:

```suggestion
        auto cs = ctx_->call_stack;
        require_eq(cs.size(), 4u);
        check_eq(cs[0]->description(),
                 "scenario with <foo> syntax in description");
        check_eq(cs[1]->description(), "given a <description>");
        check_eq(cs[2]->description(), "when <parsing> the raw description");
        check_eq(cs[3]->description(),
                 "then we copy the <description> verbatim");
```

**Review by Neverlord — CHANGES_REQUESTED** [review]

File: libcaf_test/caf/test/outline.hpp [hunk 4]
```diff
@@ -11,7 +11,7 @@
 
 namespace caf::test {
 
-class CAF_TEST_EXPORT outline : public runnable {
+class CAF_TEST_EXPORT outline_setter : public runnable {
```
Comment (Neverlord) [inline libcaf_test/caf/test/outline.hpp:14]: This is not a setter. This is the object that will actually run the tests (examples).

I'd prefer a name like `runnable_with_examples` to have a better description of what it actually is, plus please move this to its own file.

Comment (riemass) [reply libcaf_test/caf/test/outline.hpp]: Nice suggestion, will do.

**Review by Neverlord — APPROVED** [review]
